src/viam_mlmodelservice_triton_impl.cpp

```diff
@@ -957,23 +905,22 @@ class Service : public vsdk::MLModelService, public vsdk::Stoppable, public vsdk
         return nullptr;
     }
 
-    cxxapi::unique_ptr<TRITONSERVER_InferenceRequest> get_inference_request_(
-        const std::shared_ptr<struct state_>& state) {
+    cxxapi::unique_ptr<TRITONSERVER_InferenceRequest> get_inference_request_() {
```

Comment:
Arguably, `get_inference_request` and `release_inference_request` could become methods on `struct state`, but for now it doesn't really have any behavior so I've chosen not to do that. The upside to doing that would be that the `mutex` guarding `inference_requests` could be private, making it clear that only `state` deals with locking it.